packages/boot/src/boot-strapper.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  Context,
+  inject,
+  Constructor,
+  Binding,
+  BindingScope,
+  resolveList,
+} from '@loopback/context';
+import {
+  Booter,
+  BootOptions,
+  BOOTER_PHASES,
+  CoreBindings,
+  Application,
+} from '@loopback/core';
+import {resolve} from 'path';
+import {BootBindings} from './keys';
+
+import * as debugModule from 'debug';
+const debug = debugModule('loopback:boot:bootstrapper');
+
+export class BootStrapper {
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE) private app: Application,
+  ) {}
+
+  /**
+   * Function is responsible for calling all registered Booter classes that
+   * are bound to the Application instance. Each phase of an instance must
+   * complete before the next phase is started.
+   * @param {BootOptions} bootOptions Options for boot. Bound for Booters to
+   * receive via Dependency Injection.
+   */
+  async boot(bootOptions: BootOptions, ctx?: Context): Promise<Context> {
```

Comment:
`ctx` is not documented in tsdoc above. It makes me wonder if it's a good idea to allow callers to pass a custom context - what if the passed `ctx` is not inheriting from the `app`? Will everything keep working?

Reply:
In the future, the `Bootstrapper` should be able to boot artifacts from a component module. Having isolated contexts for each `boot` will be useful.

Reply:
The optional ctx requires labelling in the API docs - no magic API hacks please.
We should be very clear that it's a parameter we use for testing, rather than something to be utilized freely.
`@param [Context] ctx An optional context, used for testing and evaluation. We do not recommend using this property for production purposes.`

Alternatively, (and perhaps, preferably), we should make an optional argument for the BootOptions class that allows the passing of context in this way so that options is the only thing used to configure the behaviour of the boot function. It should still be documented with the appropriate TSDoc param tag.